### PR TITLE
Add Apiary Django model

### PR DESCRIPTION
**Pull request by mmcfarland**
## Overview

Django model and migration to add ORM model for Apiary features.

Connects #332 

## Testing Instructions

* Check out branch and run migrations
   `./scripts/manage.sh migrate`
* Interact with model in shell
  ```python
   > ./scripts/manage.sh shell
   from apps.beekeepers import models
   from django.contrib.auth.models import User

   user = User.objects.first()
   bee = models.Apiary(
       lat=34.34,
       lng=-34.342,
       marker="D",
       name="Matt's Cool Apiary",
       user=user
    )

    bee.save()


    print(bee)
    print(bee.id)
   ```
* Confirm apiary is persisted to DB

   ```shell
    > ./scripts/manage.sh dbshell
    select * from beekeepers_apiary;
    ```

### CODE REVIEW

File: src/icp/apps/beekeepers/models.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    scores = models.TextField(
+        null=True,
+        help_text='JSON representation of raster values at lat/lng')
+    starred = models.BooleanField(
+        default=False,
+        help_text='Has the apiary been starred in the UI?')
+    surveyed = models.BooleanField(
+        default=False,
+        help_text='Has the apiary been designated as requiring surveys?')
+    updateded_at = models.DateTimeField(
```
[the remark this answers is not in the document]
Comment (mmcfarland) [reply src/icp/apps/beekeepers/models.py]: Oops, will fix but need to regenerate the migration.

File: src/icp/apps/beekeepers/models.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        max_length=255,
+        help_text='Short text identifier for apiary')
+    scores = models.TextField(
+        null=True,
+        help_text='JSON representation of raster values at lat/lng')
+    starred = models.BooleanField(
+        default=False,
+        help_text='Has the apiary been starred in the UI?')
+    surveyed = models.BooleanField(
+        default=False,
```
[the remark this answers is not in the document]
Comment (mmcfarland) [reply src/icp/apps/beekeepers/models.py]: I don't think so. I believe this is the attribute which gets set when the user designates a real apiary and clicks the clipboard icon. From a UX perspective, I think designating an opt-in fits better than potentially clogging their "needs missing survey" screen and preventing them finding the real ones.  These un-surveyed records will show up in the bottom section, with an option to add to surveyable.

File: src/icp/apps/beekeepers/models.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+class Apiary(models.Model):
+    """An Apiary represents a real or possible physical location
+    where multiple honeybee colonies are located.
+    """
+    created_at = models.DateTimeField(
+        auto_now_add=True)
+    lat = models.FloatField()
+    lng = models.FloatField()
+    marker = models.CharField(
+        max_length=255,
```
[the remark this answers is not in the document]
Comment (mmcfarland) [reply src/icp/apps/beekeepers/models.py]: I'd prefer to leave this at 255.  Borne of experience, I generally don't store char fields below this size as the overhead is low and the number of times I've had to adjust too-restrictive field sizes is high.

**Comment by mmcfarland** [comment]
Good point on required fields, will add a few more constraints.

**Comment by mmcfarland** [comment]
Added non-nullable fields to most model attributes.  Table def is now:

```                                    

Table "public.beekeepers_apiary"
   Column   |           Type           |                           Modifiers                            
------------+--------------------------+----------------------------------------------------------------
 id         | integer                  | not null default nextval('beekeepers_apiary_id_seq'::regclass)
 created_at | timestamp with time zone | not null
 lat        | double precision         | not null
 lng        | double precision         | not null
 marker     | character varying(255)   | not null
 name       | character varying(255)   | not null
 scores     | text                     | 
 starred    | boolean                  | not null
 surveyed   | boolean                  | not null
 updated_at | timestamp with time zone | 
 user_id    | integer                  | not null
Indexes:
    "beekeepers_apiary_pkey" PRIMARY KEY, btree (id)
    "beekeepers_apiary_e8701ad4" btree (user_id)
Foreign-key constraints:
    "beekeepers_apiary_user_id_3aba9238eeba3f7c_fk_auth_user_id" FOREIGN KEY (user_id) REFERENCES auth_user(id) DEFERRABLE INITIALLY DEFERRED

```

I didn't add an index to `User` since we get one for free by virtue of the FK, although that's the primary query field.

File: src/icp/apps/beekeepers/models.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (fungjj92) [reply src/icp/apps/beekeepers/models.py]: SGTM!

File: src/icp/apps/beekeepers/models.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from django.db import models
+from django.conf import settings
+
+AUTH_USER_MODEL = getattr(settings, 'AUTH_USER_MODEL', 'auth.User')
```
Comment (rajadain) [inline src/icp/apps/beekeepers/models.py]: Is this in anticipation of #290?

Comment (mmcfarland) [reply src/icp/apps/beekeepers/models.py]: Not exactly, I just intended this to be a traditional FK to the User table.  I reused the mechanism being used in other models, which appears to be a best practice for this version of Django.

https://github.com/project-icp/bee-pollinator-app/blob/develop/src/icp/apps/core/models.py#L8-L12

**Pull request merged**